### PR TITLE
Sign out requested

### PR DESCRIPTION
**Pull request by rwoodnz**
Google oauth doesn’t allow query parameters on a return login or logout url. So have added a config bool to remove them.

signoutRequested is back, navigating to a return url according to configuration options previously added.

### CODE REVIEW

File: src/Auth/Protocol/OAuth.elm [hunk 1]
```diff
@@ -11,6 +11,7 @@ import OAuth
 import OAuth.AuthorizationCode as OAuth
 import Process
 import SHA1
+import Share.Config exposing (isDev)
```
Comment (supermario) [inline src/Auth/Protocol/OAuth.elm:13]: Think this needs to be removed?

File: src/Auth/Protocol/OAuth.elm [hunk 2]
```diff
@@ -105,9 +107,16 @@ initiateSignin sessionId baseUrl isDev config asBackendMsg now backendModel =
 generateSigninUrl : Url -> Auth.Common.State -> Auth.Common.ConfigurationOAuth frontendMsg backendMsg frontendModel backendModel -> Url
 generateSigninUrl baseUrl state configuration =
     let
+        queryAdjustedUrl =
+            if configuration.allowLoginQueryParameters then
+                baseUrl
+
+            else
+                { baseUrl | query = Nothing }
```
Comment (supermario) [inline src/Auth/Protocol/OAuth.elm:115]: I'm not sure if I'm a fan of this being transparently dropped, can see someone tripping up on that not being able to figure out why their params are being cleared. Okay for now but maybe there's a nice way of lifting this up to the config level in future.

**Review by supermario — APPROVED** [review]
Hey @rwoodnz – sorry I completely forgot about this!

A few comments but looks okay to me. Will merge after the `Share.Config` fix.

On my list to revisit holistically re-integrating against Dashboard in future.

Thanks 👍

File: src/Auth/Protocol/OAuth.elm (hunk 2 above)
Comment (rwoodnz) [reply src/Auth/Protocol/OAuth.elm:115]: IIRC the issue was that with google login query parameters can't be used, so added this configuration option and set it as false for google login.

File: src/Auth/Protocol/OAuth.elm (hunk 1 above)
Comment (rwoodnz) [reply src/Auth/Protocol/OAuth.elm:13]: Yep, I think you mentioned this earlier.

**Comment by rwoodnz** [comment]
I had a bug with Auth0 callback logout where CheckSession was being kicked off - not sure whether by the backend subscription or frontend shared init but appeared to be the former - and was returning a user after they should no longer exist.
So just did another commit here where I removed the clearing of the session in signoutrequested, and depend on calling the AuthLogoutRequested (configured logout) separately from frontend to remove session
~~from the front end at the point the user clicks SignOut.~~
~~In my case I then have a dev delay and then call the signoutrequested for the navigation.~~
In my case I now do a round circuit - call AuthLogoutRequested from logoutcallback and then that sends a msg back to front end that session has been removed, so user can be removed and busy spinner stopped.  
This appears to have solved my issue.

In a nutshell - with Auth0 logins we need to do Auth0's signout first, then come back to our callback before removing our lamdera session. Removing the landera session prior wasn't reliable since I think navigation changes were causing Auth0 session to be re-recognised.

File: src/Auth/Protocol/OAuth.elm (hunk 2 above)
Comment (rwoodnz) [reply src/Auth/Protocol/OAuth.elm:115]: Added a code comment to this effect

**Comment by supermario** [comment]
Sounds rather tricksy @rwoodnz ! Wonder if we can better visualise all this bouncing around somehow, if its complex for us it'll be even more so for users.

Anyway, good addition for now, merging in and lets see how goes :)
